includes/models/model.llms.user.certificate.php

```diff
@@ -114,6 +125,100 @@ public function can_user_view( $user_id = null ) {
 
 	}
 
+	/**
+	 * Retrieve information about the certificate background image.
+	 *
+	 * @since [version]
+	 *
+	 * @return array {
+	 *     Returns an associative array of information about the background image.
+	 *
+	 *     @type string $src        The image source url.
+	 *     @type int    $width      The image display width, in pixels.
+	 *     @type int    $height     The image display height, in pixels.
+	 *     @type bool   $is_default Whether or not the default image was returned.
+	 * }
+	 */
+	public function get_background_image() {
+
+		$id = $this->get( 'id' );
+
+		// Don't retrieve a size if legacy mode is enabled.
+		$size = llms_parse_bool( get_option( 'lifterlms_certificate_legacy_image_size', 'yes' ) ) ? 'full' : 'lifterlms_certificate_background';
```

Comment:
We used to have a lot of users *complain* that the certificate sizing was tied to the size of the uploaded image. It was constructed this way initially so that users *didn't have to worry* about what size to make their image. Instead they could upload their image at whatever size they'd prefer and it would automatically set the size based on the image size. But over time it resulted it a lot of users saying "I DON'T KNOW WHAT SIZE TO MAKE THE IMAGE" and I'd have to continually explain (and link to documentation) explaining how it worked.

My solution -- which does seem dumb today -- was conceived so that users could control the image size with an option in favor of being able to control it with the image.

It's all maybe a dumb conversation because what we actually should be doing is controlling the size of the canvas based on common print dimensions (and using their equivalent pixels on screen) and stretching the background image to fill the space at 100% width and height.

Possibly we could rewrite this (now) to have a default width (selected from a dropdown of predetermined print sizes (A4, etc...) and also having that same option available on a per-certificate setting. Default used if no override is present.

Perhaps converting the existing options to a "custom" size (enabled by default via a migration) and calling the "legacy" sizing "automatic" sizing (and enable that by default with a migration if it's enabled).

maybe... I don't know...

My memory is also terrible perhaps I'm remembering all of this wrong.

Reply:
Alright, we can _fix_ this later, let's think about this a bit more...

Reply:
I got thinking about how to implement and it might not be so hard...

But the thing that was hurting my brain is something you addressed at some point and I can't find... The problem is that paper sizes don't equate well with pixels. Pixels are relative to the DPI whereas paper size is absolute.

So if we think about the certificate sizing option to be the size of the page / certificate canvas (rather than as the size of the background image) and we center the background image vertically and horizontally (and maybe stretch it to fit?) that's all cool... but we shouldn't use pixels, we should use the actual dimensions of the paper. Which we can actually do in CSS (though I don't have a lot of experiencing writing screens in inches / mm via CSS)...

So we could build a dropdown with some default common paper sizes (this would probably be good to start, we can filter it and if we get requests for other sizes just add them in) and then have another option to select orientation (portrait vs landscape).

The existing sizing option could map to the "CUSTOM" option which would be populated with the existing pixels stored in the DB. But then my head breaks... If we want to make our lives easier we should probably switch to mm/inches so then what do we have the user input their pixels? Or maybe we "break" and we don't allow them to define a custom size at all anymore... 

This array contains some common sizes we could use -- I've opted to add the dimensions as `[ mm, in ]` so that we don't have to prioritize US over the rest of the (sane) world (or vice versa). That further complicates user input if we allow them to define their own size, do they have to enter both? I guess we could have them select the unit (and generate the other unit for them automatically)?

Anyway... maybe this is something for next time....

```
	public function get_sizes() {

		return array(
			// ISO 216 sizes.
			'A3' => array(
				'name' => _x( 'A3', 'Paper size name', 'lifterlms' ),
				'width' => array( 297, 11.7 ),
				'height' => array( 420, 16.5 ),
			),
			'A4' => array(
				'name' => _x( 'A4', 'Paper size name', 'lifterlms' ),
				'width' => array( 210, 8.3 ),
				'height' => array( 297, 11.7 ),
			),
			'A5' => array(
				'name' => _x( 'A5', 'Paper size name', 'lifterlms' ),
				'width' => array( 148, 5.8 ),
				'height' => array( 210, 8.3 ),
			),
			// North American sizes.
			'LETTER' => array(
				'name' => _x( 'North American Letter', 'Paper size name', 'lifterlms' ),
				'width' => array( 216, 8.5 ),
				'height' => array( 279, 11 ),
			),
			'LEGAL' => array(
				'name' => _x( 'North American Legal', 'Paper size name', 'lifterlms' ),
				'width' => array( 216, 8.5 ),
				'height' => array( 356, 14 ),
			),
			'LEDGER' => array(
				'name' => _x( 'North American Ledger', 'Paper size name', 'lifterlms' ),
				'width' => array( 279, 11 ),
				'height' => array( 432, 17 ),
			),
			'CUSTOM' => array(
				'name' => _x( 'Custom', 'Paper size name', 'lifterlms' ),
				'width' => array( ?, ? ),
				'height' => array( ?, ? ),
			),

		);

	}
```

Reply:
paper size <=> px
we included a link in our documentation: https://lifterlms.com/docs/certificate-background-image-sizes/
But yeah I see your point.
Ok, I think what we have to do is to provide something like, for the custom size:
- dpi select (common dpis)
defaults is 96
- width
input number (allows floats) - select (in|mm|px)
- height
input number (allows floats) - select (in|mm|px)
then depending on their choice we'll be able to do our conversion, right?
Complicated? Yeah, but not our fault, instead of Custom we could call that Advanced :D